src/sas/qtgui/Perspectives/Fitting/SmearingWidget.py

```diff
@@ -146,7 +149,8 @@ def updateKernelModel(self, kernel_model=None, keep_order=False, old_index=None)
             self.setElementsVisibility(False)
             return
         # Find out if data has dQ
-        (self.smear_type, self.dq_l, self.dq_r) = self.getSmearInfo()
+        self.current_smearer = smear_selection(self.data, self.kernel_model)
+        self.getSmearInfo()
```

Comment:
Is there any reason to have `smear_type`, `dq_l` and `dq_r` as state? The only place we are using them is in populating the boxes, and the getter is only called once. I think keeping these three as local variables is slightly cleaner.

Reply:
`smear_type`, `dq_l `and `dq_r` are used when the Q-resolution combo box changes to dQ, which could happen when a user wants to check a fit without any Q resolution effects and then switches back to the original fit.